src/Nowruz/modules/contract/components/contractCard/useContractCard.tsx

```diff
@@ -20,8 +20,35 @@ export const useContractCard = (offer: Offer, mission?: Mission) => {
 
   const name = type === 'users' ? offerVal.offerer.meta.name : offerVal.recipient.meta.name;
   const profileImageUrl = type === 'users' ? offerVal.offerer.meta.image : offerVal.recipient.meta.avatar;
-  const currencyIconName = offerVal.currency === 'JPY' ? 'currency-yen-circle' : 'currency-dollar-circle';
 
+  // We might delete currency icon later (we accept only USD or JPY at the moment)
+  const currencyIconName = (() => {
+    switch (offerVal.currency) {
+      case 'JPY':
+        return 'currency-yen-circle';
+      case 'USD':
+        return 'currency-dollar-circle';
+    }
+  })();
+
+  // Format the amount depending of the currency
+  const formatCurrency = (() => {
+    switch (offerVal.currency) {
+      case 'JPY':
+         return new Intl.NumberFormat('ja-JP', {
+          useGrouping: true,
+          maximumFractionDigits: 0, // Japanese Yen typically doesn't use decimal places
+        }).format(offerVal.assignment_total);
+      case 'USD':
+          return new Intl.NumberFormat('ja-JP', {
```

Comment:
`Intl.NumberFormat('ja-JP'` @minh-d it this correct. Thanks! If it is I guess we can just put the maximumFractionDigits in condition as the rest of property are all the same.

Reply:
@bastantebom I tried a change but feel free to optimize